### PR TITLE
[Event] support set event level

### LINKED ISSUE #18153: [Event] support severity configuration
WARNING by default

### PR DESCRIPTION
**Pull request by SongGuyang**
## Why are these changes needed?
   Set event level by a new string system config "event_level".

### CODE REVIEW

**Review by scv119 — CHANGES_REQUESTED** [review]

File: src/ray/util/event.cc [hunk 1]
```diff
@@ -205,6 +205,9 @@ void RayEventContext::SetCustomFields(
 ///
 /// RayEvent
 ///
+rpc::Event_Severity RayEvent::severity_threshold_ =
```
[the remark this answers is not in the document]
Comment (rkooo567) [reply src/ray/util/event.cc:208]: Make it const?

File: src/ray/util/event.cc [hunk 2]
```diff
@@ -213,6 +216,33 @@ void RayEvent::ReportEvent(const std::string &severity, const std::string &label
...
 }
 
+bool RayEvent::IsLevelEnabled(rpc::Event_Severity event_level) {
+  return event_level >= severity_threshold_;
+}
+
+void RayEvent::SetLevel(const std::string event_level) {
+  std::string level = event_level;
+  // The environment variable have a higher priority.
+  const char *var_value = getenv("RAY_BACKEND_EVENT_LEVEL");
```
[the remark this answers is not in the document]
Comment (rkooo567) [reply src/ray/util/event.cc:226]: +1. This is not necessary I think.

[the remark this answers is not in the document]
Comment (SongGuyang) [reply src/ray/util/event.cc:226]: I found we use `RAY_BACKEND_LOG_LEVEL` to set the log level. I'm not sure why we use env var. I can remove this if don't needed.

[the remark this answers is not in the document]
Comment (SongGuyang) [reply src/ray/util/event.cc:226]: "RayConfig::instance().event_level() support setting the event_level by Ray_event_level env variable. " @scv119 Thank you for this point. I don't know this feature before.

File: src/ray/util/event.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (SongGuyang) [reply src/ray/util/event.cc:208]: Not thread-safe. We should set event level as main thread starting.

[the remark this answers is not in the document]
Comment (SongGuyang) [reply src/ray/util/event.cc:208]: If make it const, we can't write test case easily.

File: src/ray/util/event.h [hunk 3]
```diff
@@ -206,6 +208,21 @@ class RayEvent {
...
+  static bool IsLevelEnabled(rpc::Event_Severity event_level);
+
+  /// Set the event severity level.
+  ///
+  /// \param event_level The input event level. It should be one of "info", "warning",
+  /// "error" and "fatal". You can also use capital letters for the options above. And if
+  /// the enviroment variable "RAY_BACKEND_EVENT_LEVEL" has been set, event_level will be
+  /// ignored.
+  /// \return void.
+  static void SetLevel(const std::string event_level);
```
[the remark this answers is not in the document]
Comment (SongGuyang) [reply src/ray/util/event.h]: I have made it a private function, only for test.

File: src/ray/util/event.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (SongGuyang) [reply src/ray/util/event.cc]: Code updated. We should use `RayEventInit` to initialize Ray Event. This function is not thread-safe. I have added some comments.

File: src/ray/util/event.h [hunk 4]
```diff
@@ -217,15 +225,29 @@ class RayEvent {
...
  private:
   rpc::Event_Severity severity_;
   std::string label_;
   json custom_fields_;
   std::ostringstream osstream_;
 };
 
+/// Ray Event initialization.
+///
+/// This function is not thread-safe. We should use this function when main thread starts.
```
Comment (scv119) [inline src/ray/util/event.h:242]: consider using https://abseil.io/docs/cpp/guides/synchronization#call_once to ensure this is only called once.

**Review by scv119 — APPROVED** [review]

File: src/ray/util/event.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (rkooo567) [reply src/ray/util/event.cc]: This was added pretty recently haha :)

File: src/ray/util/event.h (hunk 4 above)
Comment (rkooo567) [reply src/ray/util/event.h:242]: +1. Or make some singleton logic (e.g., have a variable like "initialized" and if initialized=true, crash ray)

**Review by rkooo567 — APPROVED** [review]
Please remove the tag once addressing the last comment from @scv119

**Pull request merged**